src/components/solvers/SolverPicker.tsx

```diff
@@ -96,21 +131,31 @@ export const SolverPicker = (props: SolverPickerProps) => {
     );
   };
 
+  let selectedSolver =
+    solverChoice.requestedSolverId !== undefined
+      ? solvers.find((s) => s.id === solverChoice.requestedSolverId)
+      : null;
+
   return (
     <Container>
       {loadingSolvers ? <Text>Loading solvers...</Text> : <SolverSelection />}
 
-      {solveRequest.requestedSolverId == undefined ? (
+      {selectedSolver?.authenticationOptions !== undefined &&
+      selectedSolver?.authenticationOptions !== null
+        ? Authentication(selectedSolver.authenticationOptions)
+        : null}
+
+      {solverChoice.requestedSolverId == undefined ? (
         <SettingsView
           problemType={props.problemType}
           settingChanged={(settings) => {
-            let newSolveRequest: SolverChoice = {
-              ...solveRequest,
+            let newSolverChoice: SolverChoice = {
+              ...solverChoice,
               requestedMetaSolverSettings: settings,
             };
 
-            setSolveRequest(newSolveRequest);
-            props.setSolveRequest(newSolveRequest);
+            setSolverChoice(newSolverChoice);
+            props.setSolverChoice(newSolverChoice);
           }}
         />
       ) : null}
```

Comment:
You can use the conditional and (`&&`) operator for conditional rendering without an else block (https://react.dev/learn/conditional-rendering#logical-and-operator-)